ingress/controllers/nginx/nginx/template.go

```diff
@@ -64,6 +64,27 @@ func (ngx *Manager) loadTemplate() error {
 }
 
 func (ngx *Manager) writeCfg(cfg config.Configuration, ingressCfg IngressConfig) (bool, error) {
+	var longestName int
+	var serverNames int
+	for _, srv := range ingressCfg.Servers {
```

Comment:
> what's the tradeoff in being so precise

Using a small value means nginx cannot start or be restarted
```
[emerg] 17617#0: could not build the server_names_hash, you should increase server_names_hash_bucket_size: 32
```

> just defaulting it to something high

This depends on the amount of hostnames and FQDN length